app/src/features/apiClient/screens/PostmanImporterView/components/PostmanImporter/PostmanImporter.tsx

```diff
@@ -0,0 +1,383 @@
+    } catch (error) {
+      Logger.error("Postman data import failed:", error);
+      throw error;
+    }
+  }, [addNewEnvironment, setVariables, processedFileData.environments]);
+
+  const handleImportCollectionsAndApis = useCallback(async () => {
+    let importedCollectionsCount = 0;
+    let failedCollectionsCount = 0;
+
+    const collections = processedFileData.apiRecords.filter((record) => record.type === RQAPI.RecordType.COLLECTION);
+    const apis = processedFileData.apiRecords.filter((record) => record.type === RQAPI.RecordType.API);
+
+    const handleCollectionWrites = async (collection: RQAPI.CollectionRecord) => {
+      try {
+        const newCollection = await upsertApiRecord(
+          user?.details?.profile?.uid,
+          collection,
+          workspace?.id,
+          collection.id
+        );
+        onSaveRecord(newCollection.data);
+        importedCollectionsCount++;
+        return newCollection.data.id;
+      } catch (error) {
+        failedCollectionsCount++;
+        Logger.error("Error importing collection:", error);
+        return null;
+      }
+    };
+
+    const handleApiWrites = async (api: RQAPI.ApiRecord) => {
+      const newCollectionId = collections.find((collection) => collection.id === api.collectionId)?.id;
+      if (!newCollectionId) {
+        throw new Error(`Failed to find new collection ID for API: ${api.name || api.id}`);
+      }
+
+      const updatedApi = { ...api, collectionId: newCollectionId };
+      try {
+        const newApi = await upsertApiRecord(user.details?.profile?.uid, updatedApi, workspace?.id, updatedApi.id);
+        onSaveRecord?.(newApi.data);
+      } catch (error) {
+        failedCollectionsCount++;
+        Logger.error("Error importing API:", error);
+      }
+    };
+
+    await Promise.all([batchWrite(collections, handleCollectionWrites), batchWrite(apis, handleApiWrites)]);
+
+    if (failedCollectionsCount > 0) {
+      const failureMessage =
+        failedCollectionsCount > 0
+          ? `${failedCollectionsCount} ${
+              failedCollectionsCount > 1 ? "collections and APIs" : "collection or API"
+            } failed`
+          : "";
+      if (failureMessage.length) {
+        toast.warn(`Some imports failed: ${failureMessage}, Please contact support if the issue persists.`);
+      }
+    }
+
+    return importedCollectionsCount;
+  }, [processedFileData.apiRecords, user?.details?.profile?.uid, workspace?.id, onSaveRecord, batchWrite]);
+
+  const handleImportVariables = useCallback(async () => {
+    if (Object.keys(processedFileData.variables).length === 0) {
+      return Promise.resolve();
+    }
+    const newEnvironment = await addNewEnvironment("New Environment");
+    if (newEnvironment) {
+      return setVariables(newEnvironment.id, processedFileData.variables);
+    }
+    return Promise.resolve();
+  }, [processedFileData.variables, setVariables, addNewEnvironment]);
```

Comment:
since not being used, add a `todo` comment for this function somewhere

Reply:
This is being used